src/otx/v2/adapters/torch/lightning/anomalib/dataset.py

```diff
@@ -0,0 +1,167 @@
+"""OTX adapters.torch.lightning.anomalib.Dataset API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+from anomalib.data.base.datamodule import collate_fn
+from omegaconf import DictConfig, OmegaConf
+
+from otx.v2.adapters.torch.lightning.anomalib.modules.data.data import OTXAnomalyDataset
+from otx.v2.adapters.torch.lightning.dataset import LightningDataset
+from otx.v2.api.entities.task_type import TaskType, TrainType
+from otx.v2.api.entities.utils.dataset_utils import (
+    contains_anomalous_images,
+    split_local_global_dataset,
+)
+from otx.v2.api.utils.decorators import add_subset_dataloader
+from otx.v2.api.utils.type_utils import str_to_subset_type, str_to_task_type
+
+if TYPE_CHECKING:  # pragma: no cover
```

Comment:
can you explain the no-cover comment here

Reply:
This is a comment to ignore that TYPE_CHECKING syntax when measuring unit test coverage. What do you think?

Reply:
Maybe we could add this to our `pyproject.toml` file. Otherwise, we will have to ignore this every time we have `if TYPE_CHECKING` statement. The following link might help 
https://github.com/nedbat/coveragepy/issues/831

Thoughts?

Reply:
I see, I'll fix this.

Reply:
Great, thanks!

Reply:
I'll make this work with a PR that removes black.